apps/angular/ngfor-enhancement/src/app/app.component.ts

```diff
@@ -1,25 +1,33 @@
-import { NgFor, NgIf } from '@angular/common';
 import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { NgForEmptyDirective } from './ng-for-empty.directive';
 
 interface Person {
   name: string;
 }
 
 @Component({
   standalone: true,
-  imports: [NgFor, NgIf],
+  imports: [NgForEmptyDirective],
   selector: 'app-root',
   template: `
-    <ng-container *ngIf="persons.length > 0; else emptyList">
-      <div *ngFor="let person of persons">
-        {{ person.name }}
-      </div>
-    </ng-container>
+    <div><button (click)="addPerson()">Add</button></div>
+    <div *ngForEmpty="let person of persons; empty: emptyList">
+      {{ person.name }}
+      <button (click)="removePerson(person)">Remove</button>
+    </div>
```

Comment:
you will not need any of that if the were using the new control flow, but it's good to know that you can extend directive, or even structural directive is still very useful.